api/src/test/java/org/openmrs/api/context/ContextTest.java

```diff
@@ -82,7 +82,40 @@ public void authenticate_shouldNotAuthenticateWithNullUsername() {
 	 */
 	@Test(expected = ContextAuthenticationException.class)
 	public void authenticate_shouldNotAuthenticateWithNullUsernameAndPassword() {
-		Context.authenticate(null, null);
+		Context.authenticate((String) null, null);
+	}
+	
+	/**
+	 * @see Context#authenticate(String,String)
+	 */
+	@Test(expected = ContextAuthenticationException.class)
+	public void authenticate_shouldNotAuthenticateWithNullCredentials() {
+		Context.authenticate((AuthenticationScheme) null  , null);
+	}
+	
+	/**
+	 * @see Context#authenticate(AuthenticationScheme, Credentials)
+	 */
+	@Test(expected = ContextAuthenticationException.class)
+	public void authenticate_shouldThrowWhenUnfitCredentials() {
+		Context.authenticate(new UserPassAuthenticationScheme(), new TestUuidCredentials("c1d8f5c2-e131-11de-babe-001e378eb67e"));
+	}
+	
+	/**
+	 * @see Context#authenticate(AuthenticationScheme, Credentials)
+	 */
+	@Test
+	public void authenticate_shouldAuthenticateUserThoughAlternateScheme() {
```

Comment:
@ahammi super minor change here, typo left in the method name: `authenticate_shouldAuthenticateUserThoughAlternateScheme` → `authenticate_shouldAuthenticateUserWithAlternateScheme`.